src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -883,138 +831,104 @@ protected void workerDone(Worker worker) {
+     * queue.
      * 
-     * @param task Task to execute once we have an available worker
-     * @throws InterruptedException Thrown if thread is interrupted while waiting for a worker
+     * @param task Task to insert into the schedule queue
      */
-    protected void runHighPriorityTask(TaskWrapper task) throws InterruptedException {
-      Worker w = null;
-      synchronized (workersLock) {
-        if (! shutdownFinishing) {
-          if (currentPoolSize >= maxPoolSize) {
-            // set this value initially so that incoming low priority tasks will recognized the needed delay
-            lastHighDelayMillis = task.getDelayEstimateInMs();
-            // we can't make the pool any bigger
-            w = getExistingWorker(Long.MAX_VALUE);
-            // we re-set this value to include the time it took to get a thread
-            lastHighDelayMillis = task.getDelayEstimateInMs();
-          } else {
-            lastHighDelayMillis = 0;
-            
-            if (availableWorkers.isEmpty()) {
-              w = makeNewWorker();
-            } else {
-              // always remove from the front, to get the newest worker
-              w = availableWorkers.removeFirst();
-            }
-          }
+    public void addScheduled(TaskWrapper task) {
+      int insertionIndex;
+      synchronized (scheduleQueue.getModificationLock()) {
+        ClockWrapper.stopForcingUpdate();
+        try {
+          insertionIndex = ListUtils.getInsertionEndIndex(scheduleQueue, task, true);
+          
+          scheduleQueue.add(insertionIndex, task);
+        } finally {
+          ClockWrapper.resumeForcingUpdate();
         }
       }
       
-      if (w != null) {  // may be null if shutdown
-        w.nextTask(task);
+      if (insertionIndex == 0) {
+        LockSupport.unpark(runningThread);
       }
     }
-  
+
     /**
-     * After a task has been pulled from the queue and is ready to execute it is provided here.  
-     * This function will get an available worker, waiting a bit of time for one to become 
-     * available if none are immediately available.  If after that there is still none available it 
-     * will create one (assuming we have not reached our max pool size).  Then the acquired worker 
-     * will be provided the task to execute.
+     * Call to find and reposition a scheduled task.  It is expected that the task provided has 
+     * already been added to the queue.  This call will use 
+     * {@link RecurringTaskWrapper#getNextDelayInMillis()} to figure out what the new position 
+     * within the queue should be.
      * 
-     * @param task Task to execute once we have an available worker
-     * @throws InterruptedException Thrown if thread is interrupted while waiting for a worker
+     * @param task Task to find in queue and reposition based off next delay
      */
-    protected void runLowPriorityTask(TaskWrapper task) throws InterruptedException {
```

Comment:
This is where the diff gets confusing...it's still showing removals of code that used to be in the "WorkerPool"....the worker pool logic is much simpler now, it's all in the queue handler and the queue set.